internal/engine/experiment.go

```diff
@@ -277,6 +179,79 @@ func (e *experiment) OpenReportContext(ctx context.Context) error {
 	return nil
 }
 
+// MeasureWithContext implements [model.Experiment].
+func (e *experiment) MeasureWithContext(ctx context.Context, input string) (*model.Measurement, error) {
+	// Here we ensure that we have already looked up the probe location
+	// information such that we correctly populate the measurement and also
+	// VERY IMPORTANTLY to scrub the IP address from the measurement.
+	//
+	// Also, this SHOULD happen before wrapping the context for byte counting
+	// since MaybeLookupLocationContext already accounts for bytes I/O.
+	//
+	// TODO(bassosimone,DecFox): historically we did this only for measuring
+	// and not for opening a report, which probably is not correct. Because the
+	// function call is idempotent, call it also when opening a report?
+	if err := e.session.MaybeLookupLocationContext(ctx); err != nil {
```

Comment:
I was wondering if `MaybeLookupLocationWithContext` would be more descriptive?

Re. moment to call it, a subtle concern here might be the low likelihood that there's a network change between measurement and submission. 

Thinking mostly in asynchronous submission, I think it'd be healthy to a) call it when submitting; b) raise an error (or warning) if submitting a msmt for a different (CC,ASN) that the report we've opened. I can work on that as part of my future deliverables, that include deferred submission (which is an special case).

Reply:
@ainghazal wrote:

> I was wondering if `MaybeLookupLocationWithContext` would be more descriptive?

Yeah, so, first of all we can gain some more characters by removing `WithContext`, which is an artifact from when we also had a version of the call without a context. How would you suggest to change the name?

> Re. moment to call it, a subtle concern here might be the low likelihood that there's a network change between measurement and submission.
>
> Thinking mostly in asynchronous submission, I think it'd be healthy to a) call it when submitting; b) raise an error (or warning) if submitting a msmt for a different (CC,ASN) that the report we've opened. I can work on that as part of my future deliverables, that include deferred submission (which is an special case).

As far as I know, when we're resubmitting we're using a `probeservices.Submitter`: https://github.com/ooni/probe-cli/blob/master/internal/probeservices/collector.go#L185. The code should open reports only based on the data stored inside the measurement, so it should support the use case of measuring, say, w/o VPN, and then submitting with VPN. It would be great to have another pair of eyes looking at the code as well as having some manual testing to ensure whether this is still true. It might also be good to have a test guaranteeing that this is the case.

Reply:
>  How would you suggest to change the name?
if so, perhaps just `MaybeLookupLocation`?

Reply:
Sure!